modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1285,3 +1285,103 @@ def finalize(cls, partitions):
             Partitions of Modin Dataframe on which all deferred calls should be performed.
         """
         [part.drain_call_queue() for row in partitions for part in row]
+
+    @classmethod
+    def repartition(cls, partitions, axis, lengths, widths):
+        """
+        Repartition dataframe.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            A 2D NumPy array of partitions to repartition.
+        axis : {0, 1}
+            Repartition axis.
+        lengths : list
+            The length of each partition in the rows. The "height" of
+            each of the block partitions.
+        widths : list
+            The width of each partition in the columns. The "width" of
+            each of the block partitions.
+
+        Returns
+        -------
+        tuple
+            Tuple of (new_partitions, new_lengths, new_widths)
+        """
+
+        def compute_schema(axis_lengths, axis):
+            """
+            Compute repartition schema.
+
+            The schema is a sequence of intervals in which the partitions to be combined are defined.
+            The interval is determined by the chunksize.
+
+            Parameters
+            ----------
+            axis_lengths : list
+                The length/width of each partition in the rows/columns.
+            axis : {0, 1}
+                Axis for creating the repartition scheme.
+
+            Returns
+            -------
+            tuple
+                Tuple of (schema, updated axis lengths aсcording to the schema)
+            """
+            idx = pandas.RangeIndex(sum(axis_lengths))
```

Comment:
What if `axis_lengths` is None? If you look at concat in PandasDataframe, you will see that axis_lengths can be None if axis is neither 0 or 1.

Reply:
`axis` variable in `concat` can only take the values from `{0, 1}`, can't it?

Reply:
Yes, `axis` have to be 0 or 1 but there are other conditions that can lead to axis_lengths=None.

Reply:
Indeed, will fix

Reply:
@YarShev the fix is ready